lib/generator/message.js

```diff
@@ -36,13 +36,14 @@ module.exports = function writeMessage (json, delimiters) {
       element = messageDef[groupName].elements[i];
 
       if (element.segment) {
-        if (groupJSON[element.segment]) {
+        const jsonKey = element.jsonKey || element.segment
+        if (groupJSON[jsonKey]) {
 
-          if (parseInt(element.maxOccurs === 1) || !(groupJSON[element.segment] instanceof Array)) {
-            ret += self.writeSegment(groupJSON[element.segment], element.segment, delimiters);
+          if (parseInt(element.maxOccurs === 1) || !(groupJSON[jsonKey] instanceof Array)) {
```

Comment:
@x-aaron-moore fyi, this pr https://github.com/RedoxEngine/redox-hl7-v2/pull/2 mentions a bug where `parseInt` is not done properly here. maybe this is creating further issues.